File: server/app/models/messages.py
from enum import StrEnum
from typing import Optional

from pydantic import BaseModel


class Role(StrEnum):
    USER = "user"
    ASSISTANT = "assistant"


class Message(BaseModel):
    role: Role
    content: str


class MessagePayload(BaseModel):
    thread_id: Optional[str]
    message: Message

File: frontend/src/lib/types/messages.ts
import { z } from "zod";

export const roleEnum = z.enum(["user", "assistant"]);

export const messageSchema = z.object({
  role: roleEnum,
  content: z.string(),
});

export type Message = z.infer<typeof messageSchema>;

export const messagePayloadSchema = z.object({
  thread_id: z.string().nullable(),
  message: messageSchema,
});

export type MessagePayload = z.infer<typeof messagePayloadSchema>;
